_posts/2019-10-07-a-new-vue.md

```diff
@@ -0,0 +1,92 @@
+---
+title:  "A New Vue: A Look Back at RiotJS and Looking forward to VueJS"
+date: 2019-10-08 00:00:00 +0800
+author: ted
+---
+For years, our front-end has relied on a little-known Javascript library, RiotJS. It’s lightweight, concise, and syntactically pleasant. Riot uses the `*.tag` file, and these contain the tag name, HTML, Javascript, and CSS styling. That means the general structure of a component might look something like this…
+<!--more-->
+
+```
+<hello-world>
+    <div class=”hello-world”>Hello World</div>
+<script></script>
+<style>
+  .hello-world {
+      color: blue
+}
+</style>
+</hello-world>
+```
+Like many modern JS libraries, these components can be plugged into other parts of your website depending on your needs. RiotJS worked well for us because it interacted cleanly with the rest of our stack, namely Django and rarely got in the way.
+
+That’s not to say RiotJS doesn’t have its own problems. My biggest contention with Riot was its lack of quality documentation and its small user-base. This lead to issues where finding answers on StackOverflow might yield a result that was similar to the problem I was having, but added a layer of abstraction that I couldn’t always wrap my mind around. I know “Read the docs” is a common refrain among programmers, and for good reason. Documentation gives programmers a path to follow before they turn to StackOverflow, and a reference to pull from if they do get stuck. But…
+
+I, and I’m sure many other programmers, have run into a library they decided to use where the documentation was less than stellar. RiotJS falls squarely into this category, especially as Riot upgrades from v3 to v4 and leaves the [v3 documentation](https://v3.riotjs.now.sh/api/) buried deep in their [v4 documentation](https://riot.js.org/documentation/). This was among our primary reasons for moving away from RiotJS and finding a different Javascript Library.
+
+## To Vue, React, or Elsewhere?
+From the time I got into web development in 2016 to today, Javascript libraries have encapsulated the front-end ecosystem. I learned front-end development with jQuery, but even as I was learning it was considered dated (although certainly not irrelevant) technology. Modern front-end libraries like React, Angular, Vue, and the list goes on, have overtaken jQuery and vanilla JS as the primary means by which front-end developers  interact with their codebase.
```

Comment:
vuejs logo could go here, maybe a "vuejs vs reactjs" 2min photoshop job if you think it's worth it